### PR TITLE
Update README to include more details

### LINKED ISSUE #179: Update README to include more details
Currently, our [README](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost#readme) does not provide enough details on what the provider is, what it does and what are the use cases. Add more details to make the provider more user friendly.

### PR DESCRIPTION
**Pull request by anusha94**
**What this PR does / why we need it**:
Adding more details on the BYOH Glossary, what the provider is, and reorganized few sections.
Removed the compatibility matrix, because I think the k8s compatibility is going to get more tedious to update when we move to `major.minor.*` versions. Once we tackle that, maybe then we could have a compatibility matrix listed. As of now, its just one version, so added that in text instead.

Fixes #179

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,11 +23,31 @@
...
 declarative, Kubernetes-style APIs to cluster creation, configuration and
 management.
 
-__BYOH__ is a Cluster API v1beta1 Infrastructure Provider for already-provisioned hosts running Linux.
+__BYOH__ is a Cluster API Infrastructure Provider for already-provisioned hosts running Linux. This provider allows operators to adopt Cluster API for deploying and managing kubernetes nodes without also having to adopt a specific infrastructure service. This enables users to decouple kubernetes node provisioning from host and infrastructure provisioning.
+
+## BYOH Glossary
+**Host** - A host is a running computer system. It could be physical or virtual. It has a kernel and some base operating system
+
+**BYO Host** - A host provisioned and managed outside of Cluster API
```
Comment (pshail) [inline README.md:35]:
```suggestion
**BYO Host** - A Linux host provisioned and managed outside of Cluster API
```

File: README.md [hunk 2]
```diff
@@ -71,26 +84,12 @@ More about development and contributing practices can be found in [`CONTRIBUTING
...
 ## Compatibility with Cluster API and Kubernetes Versions
 
-### Cluster API compatibility Matrix:
-
-||Cluster API v1alpha4 (v0.4)|Cluster API v1beta1 (v1.0)|
-|-|-|-|
-|BYOH Provider v1alpha1 (v0.1.0)||✓|
-
+- BYOH is currently compatible wth Cluster API v1beta1 (v1.0)
+- BYOH installer support is present and tested only for Kubernetes version `v1.22.3`. You may however use it to provision clusters of a different version, but install the Kubernetes components yourself (use the `--skip-installation` flag when starting the agent)
```
Comment (pshail) [inline README.md:88]:
```suggestion
- BYOH installer support is present and tested only for Kubernetes version `v1.22.3`. You may however use it to provision clusters of a different version by manually installing Kubernetes components (use the `--skip-installation` flag when starting the agent)
```

**Comment by pshail** [comment]
some nits, overall LGTM

**Review by dharmjit — APPROVED** [review]
lgtm!

**Pull request merged**